benchmark/benchmark_models_stepping.jl

```diff
@@ -0,0 +1,97 @@
+push!(LOAD_PATH, joinpath(@__DIR__, ".."))
+
+using BenchmarkTools
+using CUDA
+using Oceananigans
+using Benchmarks
+using Plots
+pyplot()
+# Benchmark function
+
+function benchmark_nonhydrostatic_model(Arch, FT, N)
+    grid = RectilinearGrid(Arch(), FT, size=(N, N, N), extent=(1, 1, 1))
+    model = NonhydrostaticModel(grid=grid)
+
+    time_step!(model, 1) # warmup
+
+    trial = @benchmark begin
+        @sync_gpu time_step!($model, 1)
+    end samples=10
+
+    return trial
+end
+
+function benchmark_hydrostatic_model(Arch, FT, N)
+    grid = RectilinearGrid(Arch(), FT, size=(N, N, 10), extent=(1, 1, 1))
+    model = HydrostaticFreeSurfaceModel(grid=grid, free_surface=ImplicitFreeSurface())
```

Comment:
```suggestion
    model = HydrostaticFreeSurfaceModel(grid=grid, tracers=nothing, buoyancy=nothing, free_surface=ImplicitFreeSurface())
```

If we want to compare hydrostatic with nonhydrostatic (not necessary...)